### PR TITLE
Register form

### PR DESCRIPTION
**Pull request by marielapg**
# Description

- Adds validation to input fields if they don't meet criteria: eg. email being email, username and password minimum length and blocks submission if not.
- Adds validation if user credentials already exist in database
- Adds tests to show error messages populate if the registration form is submitted empty
- Confirm password validates that the password inputs are the same and blocks submission if they are not

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x]  Testing the error validations for email, username, password, confirm password
Tested the length of the username and password

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/components/auth/Register.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import GroaWhite from "./GroaWhite.png";
 // Navbar Register
 import RegisterNavLinks from "../layout/nav-layouts/RegisterNavLinks";
 
+
```
Comment (cmruss) [inline src/components/auth/Register.js]: Empty space could be cleaned up /removed

File: src/components/auth/register.scss [hunk 2]
```diff
@@ -0,0 +1,112 @@
+.container{
```
Comment (cmruss) [inline src/components/auth/register.scss]: This file has been deleted in master, we can remove it as _register.scss is used for styling

File: src/components/dashboard/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (cmruss) [inline src/components/dashboard/index.js]: another file that has been deleted in master, we can also safely remove this as the components now live on App.js

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+REACT_APP_ANALYTICS_ID=UA-159070324-1
```
Comment (bgingrich) [inline .env]: Should gitignore the .env file.

File: src/components/auth/Register.js [hunk 5]
```diff
@@ -48,9 +48,9 @@ class Register extends React.Component {
     this.setState({ submitted: true });
     if (
       this.state.user.email &&
```
Comment (briannakeune) [inline src/components/auth/Register.js]: This regular expression will help with the @ symbol. `/^\S+@\S+$/i`

Comment (briannakeune) [reply src/components/auth/Register.js]: Will make a bug card in the future for this to update styling for validation.

**Pull request merged**